runtime/openjdk.cs

```diff
@@ -3922,19 +3922,18 @@ private static int Compare(System.Net.NetworkInformation.NetworkInterface ni1, S
             private static int GetIndex(System.Net.NetworkInformation.NetworkInterface ni)
             {
                 System.Net.NetworkInformation.IPInterfaceProperties ipprops = ni.GetIPProperties();
-                System.Net.NetworkInformation.IPv4InterfaceProperties ipv4props = ipprops.GetIPv4Properties();
-                if (ipv4props != null)
+                if (ni.Supports(System.Net.NetworkInformation.NetworkInterfaceComponent.IPv4))
                 {
+                    System.Net.NetworkInformation.IPv4InterfaceProperties ipv4props = ipprops.GetIPv4Properties();
                     return ipv4props.Index;
                 }
-                else if (InetAddressImplFactory.isIPv6Supported())
+                else if (InetAddressImplFactory.isIPv6Supported()
```

Comment:
Why not relying here only on ni.Support(...)?

Reply:
those seem to be different things
InetAddressImplFactory.isIPv6Supported() seems to be some mono related stuff

Reply:
Ok, AFAIK, from a company perspective, we don't need to support mono and no plan to support it.
On the other hand, since IKVM is released as public, it makes sense to get it :)